product-scenarios/1-manage-users-roles/1.1-user-registration/1.1.2-user-registration-by-admin/1.1.2.1-user-registration-through-apis/1.1.2.1.1-scim-1.1/src/test/java/org/wso2/identity/scenarios/test/scim/ProvisionUserWithMalformedRequestTestCase.java

```diff
@@ -0,0 +1,110 @@
+
+import org.apache.http.HttpHeaders;
+import org.apache.http.HttpResponse;
+import org.apache.http.HttpStatus;
+import org.apache.http.client.methods.HttpGet;
+import org.apache.http.client.methods.HttpPost;
+import org.apache.http.entity.StringEntity;
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.apache.http.impl.client.HttpClients;
+import org.apache.http.util.EntityUtils;
+import org.json.simple.JSONArray;
+import org.json.simple.JSONObject;
+import org.json.simple.JSONValue;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+import org.wso2.identity.scenarios.commons.ScenarioTestBase;
+import org.wso2.identity.scenarios.commons.util.Constants;
+
+import java.io.IOException;
+
+import static org.testng.Assert.assertEquals;
+import static org.testng.AssertJUnit.assertNotNull;
+
+/**
+ * This test class is to provision a user
+ * with malformed request using SCIM1.1 API.
+ */
+public class ProvisionUserWithMalformedRequestTestCase extends ScenarioTestBase {
+
+    private CloseableHttpClient client;
+    private String scimEndpoint;
+
+    @BeforeClass(alwaysRun = true)
+    public void testInit() {
+
+        setKeyStoreProperties();
+        client = HttpClients.createDefault();
+    }
+
+    /**
+     * Provision a new user.
+     * User name attribute is not the required attribute.
+     */
+    @Test(description = "1.1.2.1.1.2")
+    public void testMalformedSCIMUserCreate() throws Exception {
+
+        String USER_NAME_ATTRIBUTE = "Name";
+
+        scimEndpoint = getDeploymentProperties().getProperty(IS_HTTPS_URL) + SCIMConstants.URL_PATH_SEPARATOR +
+                Constants.SCIMEndpoints.SCIM1_ENDPOINT + SCIMConstants.URL_PATH_SEPARATOR + Constants.SCIMEndpoints.SCIM_ENDPOINT_USER;
+        HttpPost request = new HttpPost(scimEndpoint);
+        request.addHeader(HttpHeaders.AUTHORIZATION, getAuthzHeader());
+        request.addHeader(HttpHeaders.CONTENT_TYPE, SCIMConstants.CONTENT_TYPE_APPLICATION_JSON);
+
+        JSONObject rootObject = new JSONObject();
+        JSONArray schemas = new JSONArray();
+        rootObject.put(SCIMConstants.SCHEMAS_ATTRIBUTE, schemas);
+        JSONObject names = new JSONObject();
+        names.put(SCIMConstants.GIVEN_NAME_ATTRIBUTE, SCIMConstants.GIVEN_NAME_CLAIM_VALUE);
+        rootObject.put(SCIMConstants.NAME_ATTRIBUTE, names);
+        rootObject.put(USER_NAME_ATTRIBUTE, SCIMConstants.USERNAME);
+        rootObject.put(SCIMConstants.PASSWORD_ATTRIBUTE, SCIMConstants.PASSWORD);
+
+        StringEntity entity = new StringEntity(rootObject.toString());
+        request.setEntity(entity);
+
+        HttpResponse response = client.execute(request);
+        assertEquals(response.getStatusLine().getStatusCode(), HttpStatus.SC_BAD_REQUEST,
+                "User creation request is malformed hence server should have returned a bad request");
+
+        Object responseObj = JSONValue.parse(EntityUtils.toString(response.getEntity()));
```

Comment:
This section seems to be repeating. Can move to a Util or extract to a superclass.